src/linux/frida-helper-backend.vala

```diff
@@ -1988,7 +1988,11 @@ namespace Frida {
 #elif ARM
 			target_address &= ~1;
 
+# if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
```

Comment:
Does the Vala compiler define this?

Reply:
Yeah. It seems to work just fine. Without the change it dies.

Reply:
ISTM this will always be false, because I can't see the Vala compiler defining this. (Just grepped through the codebase.)

Reply:
I am setting the `CFLAGS` environment variable before I run `./configure` and `make` to set `-mbe8` and it seems to work ok for me. Without this change `frida-inject` seems to fail? Perhaps `vala` still uses the C pre-processor and that's why this works?

Reply:
Let me check this and confirm that it is working as expected.